src/P151_Honeywell_pressure.ino

```diff
@@ -0,0 +1,168 @@
+#include "_Plugin_Helper.h"
+#ifdef USES_P151
+
+// #######################################################################################################
+// ####################### Plugin 151 Honeywell Digital Output Pressure Sensors ##########################
+// #######################################################################################################
+
+
+# define PLUGIN_151
+# define PLUGIN_ID_151         151
+# define PLUGIN_NAME_151       "Environment - I2C Honeywell Pressure"
+# define PLUGIN_VALUENAME1_151 "Pressure"
+# define PLUGIN_VALUENAME2_151 "Temperature"
+
+# include "./src/PluginStructs/P151_data_struct.h"
+
+boolean Plugin_151(uint8_t function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+      Device[++deviceCount].Number           = PLUGIN_ID_151;
+      Device[deviceCount].Type               = DEVICE_TYPE_I2C;
+      Device[deviceCount].VType              = Sensor_VType::SENSOR_TYPE_DUAL;
+      Device[deviceCount].Ports              = 0;
+      Device[deviceCount].PullUpOption       = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption      = true;
+      Device[deviceCount].ValueCount         = 2;
+      Device[deviceCount].SendDataOption     = true;
+      Device[deviceCount].TimerOption        = true;
+      Device[deviceCount].TimerOptional      = true;
+      Device[deviceCount].GlobalSyncOption   = true;
+      Device[deviceCount].PluginStats        = true;      
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_151);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_151));
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_151));
+      break;
+    }
+
+    case PLUGIN_I2C_HAS_ADDRESS:
+    case PLUGIN_WEBFORM_SHOW_I2C_PARAMS:
+    {
+      // 8 bit I2C address notation in datasheet,
+      // default: 40 (28 hex).
+      // Other available standard addresses are:
+      // 56 (38 hex)
+      // 72 (48 hex)
+      // 88 (58 hex)
+      // 104 (68 hex)
+      // 120 (78 hex)
+      // 136 (88 hex)
+      // 152 (98 hex)
```

Comment:
Aren't these addresses going to be 'flattened' to `0x08` and `0x18`, as these I2C addresses can only be 7 bit wide...? And also they can't be properly stored in an `int8_t` that is often used to store the _used_ address, so -1 can be used to indicate None/Unused?

Reply:
The I2C helper functions like `addFormSelectorI2C` and `intArrayContains` accept `uin8_t`.

These adresses are how they also show up in the I2C scan and are noted in the datasheet and product nomenclature.